test/dns_dnscmd/dnscmd_test.rb

```diff
@@ -13,51 +15,51 @@ def test_dnscmd_provider_initialization
   end
 
   def test_create_address_record
-    Proxy::Dns::Dnscmd::Record.any_instance.expects(:dns_find).with('host.domain').returns(false)
-    Proxy::Dns::Dnscmd::Record.any_instance.expects(:execute).with('/RecordAdd domain host.domain. A 192.168.33.33', anything).returns(true)
-    assert Proxy::Dns::Dnscmd::Record.new.create_a_record('host.domain', '192.168.33.33')
+    Proxy::Dns::Dnscmd::Record.any_instance.expects(:dns_find).with('host.foo.bar.domain.local').returns(false)
+    Proxy::Dns::Dnscmd::Record.any_instance.expects(:execute).with('/RecordAdd bar.domain.local host.foo.bar.domain.local. A 192.168.33.33', anything).returns(true)
```

Comment:
I don't understand this bit: we are adding a host 'host.foo.bar.domain.local' to 'bar.domain.local'. 'host.foo' isn't a legal host name, how would ms dns handle this -- is it going to automatically create a subdomain 'foo.bar.domain.local'? I don't think this is a desired behaviour...